lib/src/parse/parser.dart

```diff
@@ -520,21 +520,26 @@ class Parser {
   /// Consumes the next character if it's equal to [letter], ignoring ASCII
   /// case.
   @protected
-  bool scanCharIgnoreCase(int letter) {
-    if (!equalsLetterIgnoreCase(letter, scanner.peekChar())) return false;
-    scanner.readChar();
-    return true;
+  bool scanAsciiChar(int letter, {bool caseSensitive = false}) {
```

Comment:
It's confusing to me that this is called `scanAsciiChar()` when the only relationship it has to ASCII is ignoring the case of letters, and then only if `caseSensitive` is `false`. As a reader, I'd expect `scanAsciiChar($unicodeSnowman)` to be invalid, but it's totally fine.

I'd probably just call this `scanChar()`, and explain in the doc comment that you'd use it over `scanner.scanChar()` when you want the case-sensitivity to be configurable.